examples/lump-mpi.py

```diff
@@ -85,34 +86,10 @@ def main(ctx_factory=cl.create_some_context):
     box_ur = 5.0
 
     comm = MPI.COMM_WORLD
-    nproc = comm.Get_size()
     rank = comm.Get_rank()
-    num_parts = nproc
 
-    from meshmode.distributed import (
-        MPIMeshDistributor,
-        get_partition_by_pymetis,
-    )
-
-    mesh_dist = MPIMeshDistributor(comm)
-    global_nelements = 0
-    local_nelements = 0
-    if mesh_dist.is_mananger_rank():
-        from meshmode.mesh.generation import generate_regular_rect_mesh
-
-        mesh = generate_regular_rect_mesh(
-            a=(box_ll,) * dim, b=(box_ur,) * dim, n=(nel_1d,) * dim
-        )
-        global_nelements = mesh.nelements
-        logging.info(f"Total {dim}d elements: {global_nelements}")
-
-        part_per_element = get_partition_by_pymetis(mesh, num_parts)
-
-        local_mesh = mesh_dist.send_mesh_parts(mesh, part_per_element, num_parts)
-        del mesh
-
-    else:
-        local_mesh = mesh_dist.receive_mesh_part()
+    local_mesh, global_nelements = create_parallel_box_grid(comm, dim, box_ll,
+                                                            box_ur, nel_1d)
```

Comment:
I think I'd prefer something a bit more general that gets the mesh by a callback. (rather than being specific to boxes)

Reply:
Yeah, ok, try this on.   I *think* this is something like what you are getting at.